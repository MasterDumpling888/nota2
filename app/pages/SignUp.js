import React, { useState } from 'react';
import { View, Text, TextInput, Button, StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import PageBox from '../components/PageBox';
import { createUserWithEmailAndPassword } from 'firebase/auth';
import { auth } from '../firebaseConfig';

function SignUp({ navigation }) {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');

  const handleSignUp = async () => {
    if (password !== confirmPassword) {
      // Handle password mismatch
      console.error("Passwords do not match");
      return;
    }

    try {
      const userCredential = await createUserWithEmailAndPassword(auth, email, password);
      const user = userCredential.user;
      // Navigate to the next screen or show a success message
    } catch (error) {
      console.error(error);
      // Handle sign-up error
    }
  };

  return (
    <SafeAreaView style={{ flex: 1 }}>
      <PageBox title="Sign Up" onClose={() => navigation.goBack()}>
        <View style={styles.inputContainer}>
          <Text>Email</Text>
          <TextInput
            style={styles.input}
            value={email}
            onChangeText={setEmail}
            keyboardType="email-address"
            autoCapitalize="none"
          />
        </View>
        <View style={styles.inputContainer}>
          <Text>Password</Text>
          <TextInput
            style={styles.input}
            value={password}
            onChangeText={setPassword}
            secureTextEntry
          />
        </View>
        <View style={styles.inputContainer}>
          <Text>Confirm Password</Text>
          <TextInput
            style={styles.input}
            value={confirmPassword}
            onChangeText={setConfirmPassword}
            secureTextEntry
          />
        </View>
        <Button title="Sign Up" onPress={handleSignUp} />
      </PageBox>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  inputContainer: {
    marginBottom: 16,
  },
  input: {
    borderWidth: 1,
    borderColor: 'gray',
    borderRadius: 5,
    padding: 8,
    marginTop: 8,
  },
});

export default SignUp;
