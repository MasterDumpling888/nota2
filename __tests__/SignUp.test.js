import React from 'react';
import { render } from '@testing-library/react-native';
import SignUp from '../app/pages/SignUp';


describe('SignUp Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders SignUp form', () => {
    const { getByText } = render(<SignUp navigation={global.mockNavigation} />);
    
    // Check if title and form fields are rendered
    expect(getByText('Sign Up')).toBeTruthy();
    expect(getByText('Username')).toBeTruthy();
    expect(getByText('Email')).toBeTruthy();
    expect(getByText('Password')).toBeTruthy();
    expect(getByText('Confirm Password')).toBeTruthy();
  });

    // Snapshot test
    test('Sign Up snapshot test', () => {
        const { toJSON } = render(
            <SignUp navigation={global.mockNavigation} />
        );
        expect(toJSON()).toMatchSnapshot();
    });
});
