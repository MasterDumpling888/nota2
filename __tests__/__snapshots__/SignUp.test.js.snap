// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component Sign Up snapshot test 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flex": 1,
    }
  }
>
  <PageBox
    onClose={[Function]}
    title="Sign Up"
  >
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Raleway-Regular",
            "fontSize": 14,
            "textAlign": "left",
          }
        }
      >
        Username
      </Text>
      <TextInput
        onChangeText={[Function]}
        style={
          {
            "borderColor": "#33FD0A",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "white",
            "fontFamily": "Raleway-SemiBold",
            "fontSize": 14,
            "marginTop": 8,
            "padding": 8,
            "textAlign": "left",
          }
        }
        value=""
      />
    </View>
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Raleway-Regular",
            "fontSize": 14,
            "textAlign": "left",
          }
        }
      >
        Email
      </Text>
      <TextInput
        autoCapitalize="none"
        keyboardType="email-address"
        onChangeText={[Function]}
        style={
          {
            "borderColor": "#33FD0A",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "white",
            "fontFamily": "Raleway-SemiBold",
            "fontSize": 14,
            "marginTop": 8,
            "padding": 8,
            "textAlign": "left",
          }
        }
        value=""
      />
    </View>
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Raleway-Regular",
            "fontSize": 14,
            "textAlign": "left",
          }
        }
      >
        Password
      </Text>
      <TextInput
        onChangeText={[Function]}
        secureTextEntry={true}
        style={
          {
            "borderColor": "#33FD0A",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "white",
            "fontFamily": "Raleway-SemiBold",
            "fontSize": 14,
            "marginTop": 8,
            "padding": 8,
            "textAlign": "left",
          }
        }
        value=""
      />
    </View>
    <View
      style={
        {
          "marginBottom": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Raleway-Regular",
            "fontSize": 14,
            "textAlign": "left",
          }
        }
      >
        Confirm Password
      </Text>
      <TextInput
        onChangeText={[Function]}
        secureTextEntry={true}
        style={
          {
            "borderColor": "#33FD0A",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "white",
            "fontFamily": "Raleway-SemiBold",
            "fontSize": 14,
            "marginTop": 8,
            "padding": 8,
            "textAlign": "left",
          }
        }
        value=""
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#33FD0A",
          "borderRadius": 5,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Raleway-SemiBold",
            "fontSize": 16,
          }
        }
      >
        Sign Up
      </Text>
    </View>
  </PageBox>
</RNCSafeAreaView>
`;
